src/NamedTypesRegistry.cc

```diff
@@ -137,25 +160,83 @@ namespace drafter {
                 return members;
             }
 
+            Members collectMembers(const mson::TypeNames& tn) {
+                Members members;
+
+                for (mson::TypeNames::const_iterator itn = tn.begin() ; itn != tn.end() ; ++itn) {
+                    members.insert(name(*itn));
+                }
+
+                return members;
+            }
+
             Members collectMembers(const snowcrash::DataStructure* ds) {
                 return collectMembers(ds->sections);
             }
 
-            InheritanceComparator(const DataStructures& elements) {
+            Members collectMembers(const std::string& id, const MembersMap& members, Members& resolved) {
+                Members collected;
+
+                if (resolved.find(id) != resolved.end()) {
+                    return collected;
+                }
+
+                resolved.insert(id);
+
+                MembersMap::const_iterator member = members.find(id);
+                if (member == members.end()) {
+                    return collected;
+                }
+
+                for (Members::const_iterator i = member->second.begin() ; i != member->second.end() ; ++i) {
+                    Members sub = collectMembers(*i, members, resolved);
+                    collected.insert(*i);
+                    collected.insert(sub.begin(), sub.end());
+                }
+
+                return collected;
+            }
+
+            DependencyTypeInfo(const DataStructures& elements) {
 
-                // map inheritance
+                // map inheritance && element
                 for (DataStructures::const_iterator i = elements.begin() ; i != elements.end() ; ++i) {
+
+                    nameToElement[name(i->node)] = &(*i->node);
+
                     if (hasParent(i->node)) {
                         childToParent[name(i->node)] = parent(i->node);
+#ifdef DEBUG_DEPENDENCIES
```

Comment:
As said in other comment, please go through all occurrences of these blocks and make sure they have an empty line before and after them.